plots/laws/thermodynamics/melting_of_a_body_having_a_high_velocity.py

```diff
@@ -0,0 +1,127 @@
+    },
+    "Zn": {
+        "specific_heat_heating": 400,
+        "specific_heat_melting": 117_000,
+        "temperature_of_melting": 415,
+    }
+}
+
+velocity_of_meteorite = symbols("velocity_of_meteorite")
+temperature_of_meteorite = symbols("temperature_of_meteorite")
+koefficient_of_transfer_to_kinetic = symbols("koefficient_of_transfer_to_kinetic")
+
+specific_heat_heating_meteorite = symbols("specific_heat_heating_meteorite")
+specific_heat_melting_meteorite = symbols("specific_heat_melting_meteorite")
+temperature_of_meteorite_melting = symbols("temperature_of_meteorite_melting")
+
+mass_of_meteorite = symbols("mass_of_meteorite")
+mass_of_melting_meteorite = symbols("mass_of_melting_meteorite")
+koefficient_of_melting_meteorite = symbols("koefficient_of_melting_meteorite")
+
+energy_to_heating_meteorite = energy_heating_law.law.subs({
+    energy_heating_law.specific_heat_capacity: specific_heat_heating_meteorite,
+    energy_heating_law.body_mass: mass_of_meteorite,
+    energy_heating_law.temperature_origin: temperature_of_meteorite,
+    energy_heating_law.temperature_end: temperature_of_meteorite_melting
+}).rhs
+
+energy_to_meteorite_melting = energy_melting_law.law.subs({
+    energy_melting_law.mass_of_matter: mass_of_melting_meteorite,
+    energy_melting_law.specific_heat_melting: specific_heat_melting_meteorite
+}).rhs
+
+kinetic_energy_of_meteorite = kinetic_energy_law.law.subs({
+    kinetic_energy_law.body_mass: mass_of_meteorite,
+    kinetic_energy_law.body_velocity: velocity_of_meteorite
+}).rhs
+
+energy_conservation_equation = energy_conservation_law.law.subs({
+    energy_conservation_law.mechanical_energy(energy_conservation_law.time_before): koefficient_of_transfer_to_kinetic * kinetic_energy_of_meteorite,
+    energy_conservation_law.mechanical_energy(energy_conservation_law.time_after): energy_to_heating_meteorite + energy_to_meteorite_melting
+})
+mass_of_melting_value = solve(energy_conservation_equation, mass_of_melting_meteorite,
+    dict=True)[0][mass_of_melting_meteorite]
+koefficient_of_melting_meteorite_value = mass_of_melting_value / mass_of_meteorite
+answer = Eq(koefficient_of_melting_meteorite, koefficient_of_melting_meteorite_value)
+print(f"Total equation:\n{print_expression(answer)}")
+
+koefficient_of_melting_meteorite_to_plots = answer.subs({
+    koefficient_of_transfer_to_kinetic: 0.80,   # percents
+    temperature_of_meteorite: 300   # kelvins
+})
+
+base_plot = plot(title="The proportion of a molten meteorite depending on its velocity",
+    xlabel=r"$v, m/s$",
+    ylabel=r"$molten ratio$",
+    backend=MatplotlibBackend,
+    legend=True,
+    show=False)
+
+MASSES_RATIO_MAXIMUM = 1
+MASSES_RATIO_MINIMUM = 0
+for matter, parameters in matters_parametrs.items():
+    masses_ratio_to_subplot = koefficient_of_melting_meteorite_to_plots.subs({
+        specific_heat_heating_meteorite: parameters["specific_heat_heating"],   # joules / (kilogram * kelvin)
```

Comment:
MyPy is complaining about types here. Could you fix it?

Reply:
Convert to `float`?

Reply:
Nope. I will show how to do it.